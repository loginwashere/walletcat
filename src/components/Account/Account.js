import React, { Component, PropTypes } from 'react';

export default class Account extends Component {
  render() {
    const { account } = this.props;
    return (
      <li className="list-group-item">
        <h4 className="list-group-item-heading">
          {account.name}
        </h4>
        <p className="list-group-item-text">
          {account.description}
        </p>
      </li>
    );
  }
}

Account.propTypes = {
  account: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired
  })
}
