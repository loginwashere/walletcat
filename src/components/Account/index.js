import React, { Component, PropTypes } from 'react';
import { Link } from 'react-router';
import { Badge } from 'react-bootstrap';

export class Account extends Component {
  render() {
    const { account, accountCurrency } = this.props;
    return (
      <Link to={`/accounts/${account.id}`}
            className="list-group-item">
        <Badge bsStyle="danger" pullRight>
          {account.amount} {accountCurrency.name}
        </Badge>
        <h4 className="list-group-item-heading">
          {account.name}
        </h4>
        <p className="list-group-item-text">
          {account.description}
        </p>
      </Link>
    );
  }
}

Account.propTypes = {
  account: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    description: PropTypes.string,
    amount: PropTypes.number.isRequired
  }).isRequired,
  accountCurrency: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired
  }).isRequired
}

export default Account;
