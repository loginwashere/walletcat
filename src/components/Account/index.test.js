import React from 'react'
import renderer from 'react-test-renderer'
import Account from '.'
import {
  accountSeeder,
  userCurrencySeeder,
  currencySeeder
} from '../../../server/seeds'

describe('components:Account:', () => {
  const account = accountSeeder.items[0]
  const userCurrency = userCurrencySeeder.items
    .filter(item => item.id === account.currencyId)[0]
  const currency = currencySeeder.items
    .filter(item => item.id === userCurrency.currencyId)[0]

  it('renders without crashing', () => {
    const tree = renderer.create(
      <Account account={account}
               accountCurrency={currency} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
