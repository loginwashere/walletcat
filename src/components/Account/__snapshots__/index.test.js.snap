exports[`components:Account: renders without crashing 1`] = `
<a
  className="list-group-item"
  onClick={[Function]}
  style={Object {}}>
  <span
    className="badge badge-danger pull-right">
    0.00000000
     
    AED
  </span>
  <h4
    className="list-group-item-heading truncate">
    Wallet
  </h4>
  <p
    className="list-group-item-text truncate">
    My Wallet
  </p>
</a>
`;
